applications/FSIapplication/python_scripts/partitioned_embedded_fsi_base_solver.py

```diff
@@ -0,0 +1,665 @@
+                "input_variable_list": ["POSITIVE_FACE_PRESSURE","NEGATIVE_FACE_PRESSURE"],
+                "output_variable_list": ["DISPLACEMENT"]
+            }""")
+        else:
+            err_msg = 'Level set type is: \'' + self.level_set_type + '\'. Expected \'continuous\' or \'discontinuous\'.'
+            raise Exception(err_msg)
+
+        aux_settings["parent_model_part_name"].SetString(self.structure_interface_submodelpart_name)
+
+        # Construct the FSI coupling interface
+        fsi_coupling_interface_structure = fsi_coupling_interface.FSICouplingInterface(
+            self.model,
+            aux_settings,
+            self._get_convergence_accelerator())
+
+        KratosMultiphysics.Logger.PrintInfo('PartitionedEmbeddedFSIBaseSolver', 'Structure FSI coupling interface created')
+
+        return fsi_coupling_interface_structure
+
+    def _get_fsi_coupling_interface_fluid(self):
+        if not hasattr(self, '_fsi_coupling_interface_fluid'):
+            self._fsi_coupling_interface_fluid = self._create_fsi_coupling_interface_fluid()
+        return self._fsi_coupling_interface_fluid
+
+    def _create_fsi_coupling_interface_fluid(self):
+        # Set auxiliary settings
+        # Note that in the embedded case, the fluid interface is identical to the structure one
+        # This is intentionally done, since this copy will be used in the level set computation
+        if (self.level_set_type == "continuous"):
+            aux_settings = KratosMultiphysics.Parameters(
+            """{
+                "model_part_name": "FSICouplingInterfaceFluid",
+                "parent_model_part_name": "",
+                "input_variable_list": ["DISPLACEMENT"],
+                "output_variable_list": ["PRESSURE"]
+            }""")
+        elif (self.level_set_type == "discontinuous"):
+            aux_settings = KratosMultiphysics.Parameters(
+            """{
+                "model_part_name": "FSICouplingInterfaceFluid",
+                "parent_model_part_name": "",
+                "input_variable_list": ["DISPLACEMENT"],
+                "output_variable_list": ["POSITIVE_FACE_PRESSURE","NEGATIVE_FACE_PRESSURE"]
+            }""")
+        else:
+            err_msg = 'Level set type is: \'' + self.level_set_type + '\'. Expected \'continuous\' or \'discontinuous\'.'
+            raise Exception(err_msg)
+
+        aux_settings["parent_model_part_name"].SetString(self.structure_interface_submodelpart_name)
+
+        # Construct the FSI coupling interface
+        fsi_coupling_interface_fluid = fsi_coupling_interface.FSICouplingInterface(
+            self.model,
+            aux_settings)
+
+        KratosMultiphysics.Logger.PrintInfo('PartitionedEmbeddedFSIBaseSolver', 'Fluid FSI coupling interface created')
+
+        return fsi_coupling_interface_fluid
+
+    # This method finds the maximum buffer size between mesh,
+    # fluid and structure solvers and sets it to all the solvers.
```

Comment:
why do you need it in all solvers?

Reply:
I think it was required in the very first approach of the solver to compute the fixed mesh ALE displacement. With the current implementation I agree that it doesn't make sense at all. I'll try to remove it.

Reply:
Removed. This is not needed anymore.